docs/src/content/ignition/docs/guides/upgradeable-proxies.md

```diff
@@ -0,0 +1,372 @@
+# Using Hardhat Ignition with upgradeable proxies
+
+When developing smart contracts, you may decide to use an upgradeable proxy pattern to allow for future upgrades to your contracts. This guide will explain how to create Ignition modules to deploy and interact with your upgradeable proxy contracts.
+
+While there are several different proxy patterns, each with their own tradeoffs, this guide will focus on the [TransparentUpgradeableProxy](https://docs.openzeppelin.com/contracts/5.x/api/proxy#TransparentUpgradeableProxy) pattern. You can read more about upgradeable proxy patterns [on OpenZeppelin's blog](https://blog.openzeppelin.com/proxy-patterns).
+
+:::tip
+
+The finished code for this guide can be found in the [Hardhat Ignition monorepo](https://github.com/NomicFoundation/hardhat-ignition/tree/development/examples/upgradeable)
+
+:::
+
+## Getting to know our contracts
+
+Before we start writing our Ignition modules, let's take a look at the contracts we'll be deploying and interacting with.
+
+First, inside our `contracts` directory, we'll create a file called `Demo.sol`:
+
+```solidity
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.9;
+
+// A contrived example of a contract that can be upgraded
+contract Demo {
+  function version() public pure returns (string memory) {
+    return "1.0.0";
+  }
+}
+```
+
+This is the contract that we'll be upgrading. It's a simple contract that returns a version string.
+
+Let's go ahead and create our upgraded version of this contract in a new file called `DemoV2.sol`:
+
+```solidity
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.9;
+
+// A contrived example of a contract that can be upgraded
+contract DemoV2 {
+  string public name;
+
+  function version() public pure returns (string memory) {
+    return "2.0.0";
+  }
+
+  function setName(string memory _name) public {
+    name = _name;
+  }
+}
+```
+
+In addition to updating the version string, this contract also adds a `name` state variable and a `setName` function that allows us to set the value of `name`. We'll use this function later when we upgrade our proxy.
+
+Finally, we'll create a file called `Proxies.sol` to import our proxy contracts. This file will look a little different than the others:
+
+```solidity
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.9;
+
+import "@openzeppelin/contracts/proxy/transparent/ProxyAdmin.sol";
+import "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+```
+
+Because we're using the OpenZeppelin proxy contracts, we need to import them here to make sure Hardhat knows to compile them. This will ensure that their artifacts are available for Hardhat Ignition to use later when we're writing our Ignition modules.
```

Comment:
Do we need to add a section here suggestion the installation of `@openzeppelin/contracts`?

```sh
npm install --save-dev @openzeppelin/contracts
```

Reply:
Hmm I was thinking not, because it's mostly meant as a guide to show how it'll work with whatever proxy you choose, but on second thought, might as well add it